src/common/common-core/src/main/java/org/niis/xroad/common/core/exception/ErrorCode.java

```diff
@@ -245,17 +245,16 @@ public String code() {
     }
 
     /**
-     * Get ErrorCode from string code.
+     * Get ErrorDeviation from string code.
      *
      * @param code the string code
-     * @return the ErrorCode enum value, or INTERNAL_ERROR if not found
+     * @return the ErrorDeviation built from the code
      */
-    public static ErrorCodes fromCode(String code) {
-        for (ErrorCodes errorCode : values()) {
-            if (errorCode.code.equals(code)) {
-                return errorCode;
-            }
-        }
-        return INTERNAL_ERROR; // Default fallback
+    public static DeviationBuilder.ErrorDeviationBuilder withCode(String code) {
+        return () -> code;
+    }
+
+    public DeviationBuilder.ErrorDeviationBuilder withPrefix(String prefix) {
+        return () -> prefix + "." + code;
```

Comment:
maybe add prefix.toLowerCase(), to make sure prefix is lowercased?
there are mixed calls, like
TOKEN_NOT_FOUND.withPrefix("signer")
INTERNAL_ERROR.withPrefix("Signer")